wsaa.py

```diff
@@ -136,42 +138,60 @@ def sign_tra(tra, cert=CERT, privatekey=PRIVATEKEY, passphrase=""):
             cert = open(cert).read()
             if isinstance(cert, str):
                 cert = cert.encode("utf-8")
-        cert = x509.load_pem_x509_certificate(cert, default_backend())
+        cert = x509.load_pem_x509_certificate(cert)
 
-        try:
-            # Firmar el texto (tra) usando cryptography (openssl bindings para python)
-            p7 = _lib.PKCS7_sign(
-                cert._x509, private_key._evp_pkey, _ffi.NULL, bio_in, 0
-            )
-        finally:
-            # Liberar memoria asignada
-            _lib.BIO_free(bio_in)
-        # Se crea un buffer nuevo porque la firma lo consume
-        bio_in = _lib.BIO_new_mem_buf(tra, len(tra))
-        try:
-            # Crear buffer de salida
-            bio_out = _lib.BIO_new(_lib.BIO_s_mem())
+        if sys.version.split(" ")[0][0:3] == "2.7":
             try:
-                # Instanciar un SMIME
-                _lib.SMIME_write_PKCS7(bio_out, p7, bio_in, 0)
-
-                # Tomar datos para la salida
-                result_buffer = _ffi.new("char**")
-                buffer_length = _lib.BIO_get_mem_data(bio_out, result_buffer)
-                output = _ffi.buffer(result_buffer[0], buffer_length)[:]
+                # Firmar el texto (tra) usando cryptography (openssl bindings para python)
+                p7 = _lib.PKCS7_sign(
+                    cert._x509, private_key._evp_pkey, _ffi.NULL, bio_in, 0
+                )
+            finally:
+                # Liberar memoria asignada
+                _lib.BIO_free(bio_in)
+            try:
```

Comment:
Hi @HanslettTheDev! You have to create a new memory buffer because the signature consumes it.
```suggestion
            # Se crea un buffer nuevo porque la firma lo consume
            bio_in = _lib.BIO_new_mem_buf(tra, len(tra))
            try:
```